plugins/trace_api_plugin/store_provider.cpp

```diff
@@ -67,6 +67,7 @@ namespace eosio::trace_api {
             const auto& block = std::get<block_entry_v0>(e);
             if (block.number == block_height) {
                trace_offset = block.offset;
+               return false; // stop scanning
```

Comment:
We keep scanning until we hit lib (below) because forks can cause duplicate block numbers added to the log. We want to return the most recent and we read from front to back of the file which is appended to.

Reply:
Thank you! That makes sense.